### PR TITLE
ADA-3161 | Upgrade to PHP 8.3

### PR DESCRIPTION
**Pull request by chrisbocp**
This is a BIG upgrade!  Was previously set to PHP 5.3 and PHPUnit 4.8.  Will need some careful verification that things still work.

### CODE REVIEW

File: lib/PayPal/Formatter/FormatterFactory.php [hunk 1]
```diff
@@ -1,19 +1,17 @@
...
-            case 'NV':
-                return new PPNVPFormatter();
-                break;
-            case 'SOAP':
-                return new PPSOAPFormatter();
-                break;
-            default:
-                throw new \InvalidArgumentException("Invalid value for bindingType. You passed $bindingType");
-        }
+        return match ($bindingType) {
```
Comment (bobbysocp) [inline lib/PayPal/Formatter/FormatterFactory.php:11]: **praise:** hell yeah robot

File: lib/PayPal/Formatter/PPSOAPFormatter.php [hunk 2]
```diff
@@ -26,8 +29,9 @@ public function toString($request, $options = array())
         return $soapEnvelope . $soapHeader . $soapBody . '</soapenv:Envelope>';
     }
 
-    public function toObject($string, $options = array())
+    #[Override]
+    public function toObject($string, $options = []): never
```
Comment (bobbysocp) [inline lib/PayPal/Formatter/PPSOAPFormatter.php:33]: **praise:** never!

**Review by bobbysocp — COMMENTED** [review]
This looks pretty good. The robot did some work. Nothing looks crazy questionable.

We've marked a lot of tests defective and that's probably the right move. Still a lot running. That's a positive. Well, running for the first time here. Good job.

I don't know what the three deprecations are inside the test suite, do you?

File: composer.json [hunk 3]
```diff
@@ -12,15 +12,26 @@
         }
     ],
     "require": {
-        "php": ">=5.3.0",
+        "php": "^8.3",
```
Comment (bobbysocp) [inline composer.json:15]: **issue:** ~8.3.0 please we don't want to go to 8.4 accidentally

Comment (chrisbocp) [reply composer.json:15]: Whoops!  Got in the habit of using `^` for the Laravel stuff.

**Comment by chrisbocp** [comment]
> I don't know what the three deprecations are inside the test suite, do you?

The tests in that file are trying to use `PayPal\Auth\OAuthTokenCredential`, which doesn't exist and hasn't for some time.

**Pull request merged**